quickwit/quickwit-search/src/collector.rs

```diff
@@ -695,13 +699,149 @@ pub(crate) fn make_merge_collector(
     })
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct PartialHitSortingKey {
+    score: Option<SortValue>,
+    address: GlobalDocAddress,
+    asc: bool,
+}
+
+impl Ord for PartialHitSortingKey {
+    fn cmp(&self, other: &PartialHitSortingKey) -> Ordering {
+        assert_eq!(
+            self.asc, other.asc,
+            "comparing two PartialHitSortingKey of different ordering"
+        );
+        if self.asc {
+            self.score
+                .map(Reverse)
+                .cmp(&other.score.map(Reverse))
+                .then(other.address.cmp(&self.address))
+        } else {
+            self.score
```

Comment:
Alternatively 

`(self.score, self.address).cmp(&(other.score, other.address))`

Reply:
(nitpick)

Reply:
You can also do something like:

let descending_order =  (self.score, self.address).cmp(&(other.score, other.address));
if self.asc {
   descending_order.reverse()
} else {
   descending_order
}
```